src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/IndexOfAnyAsciiSearcher.cs

```diff
@@ -490,26 +490,23 @@ internal static int LastIndexOfAnyVectorized<TNegator>(ref byte searchSpace, int
             // X86: Downcast every character using saturation.
             // - Values <= 32767 result in min(value, 255).
             // - Values  > 32767 result in 0. Because of this we must do more work to handle needles that contain 0.
-            // ARM64: Take the low byte of each character.
-            // - All values result in (value & 0xFF).
+            // ARM64: Do narrowing saturation over unsigned values.
+            // - All values result in min(value, 255)
             Vector128<byte> source = Sse2.IsSupported
                 ? Sse2.PackUnsignedSaturate(source0, source1)
-                : AdvSimd.Arm64.UnzipEven(source0.AsByte(), source1.AsByte());
+                : AdvSimd.ExtractNarrowingSaturateUpper(AdvSimd.ExtractNarrowingSaturateLower(source0.AsUInt16()), source1.AsUInt16());
 
             Vector128<byte> result = IndexOfAnyLookupCore(source, bitmapLookup);
 
-            // On ARM64, we ignored the high byte of every character when packing (see above).
-            // The 'result' can therefore contain false positives - e.g. 0x141 would match 0x41 ('A').
             // On X86, PackUnsignedSaturate resulted in values becoming 0 for inputs above 32767.
             // Any value above 32767 would therefore match against 0. If 0 is present in the needle, we must clear the false positives.
             // In both cases, we can correct the result by clearing any bits that matched with a non-ascii source character.
-            if (AdvSimd.Arm64.IsSupported || TOptimizations.NeedleContainsZero)
+            if (TOptimizations.NeedleContainsZero)
             {
+                Debug.Assert(Sse2.IsSupported);
                 Vector128<short> ascii0 = Vector128.LessThan(source0.AsUInt16(), Vector128.Create((ushort)128)).AsInt16();
                 Vector128<short> ascii1 = Vector128.LessThan(source1.AsUInt16(), Vector128.Create((ushort)128)).AsInt16();
-                Vector128<byte> ascii = Sse2.IsSupported
-                    ? Sse2.PackSignedSaturate(ascii0, ascii1).AsByte()
-                    : AdvSimd.Arm64.UnzipEven(ascii0.AsByte(), ascii1.AsByte());
+                Vector128<byte> ascii = Sse2.PackSignedSaturate(ascii0, ascii1).AsByte();
```

Comment:
Nit: does the ascii local temporary help with anything?

Reply:
I don't think so, I just found it a bit more readable when initially writing this logic